docs: updating docker-compose docs (#4252)
jina-ai/jina: docs/advanced/experimental/docker-compose.md

(docker-compose)=
# Jina with Docker Compose

Jina is a cloud native neural search framework. Therefore, one of the simplest ways of either prototyping or serving in
production is to deploy your `Flow` with `docker-compose`.

Remember that a `Flow` defines complex processing pipelines. A `Flow` is composed of `Executors` which run python code
defined to operate on `DocumentArray`. These `Executors` will live in different runtimes depending on how you want to deploy
your Flow. By default, if you are serving your Flow locally they will live within processes. Nevertheless, 

> because Jina is thought to be cloud native your Flow can easily manage executors that live in containers and that are

Reviewer: ```suggestion
because Jina is thought to be cloud native your Flow can easily manage Executors that live in containers and that are
```

(docker-compose)=
# Jina with Docker Compose

Jina is a cloud native neural search engine. Therefore, one of the simplest way of either prototyping or serving in
production is to deploy your `Flow` with `docker-compose`.

Remember that a `Flow` define complexes processing pipelines. A `Flow` is composed of `Executors` which run python code

> define to operate on `Document`. These `Executors` will live in different runtime depending on how you want to deploy

Reviewer: ```suggestion
defined to operate on `DocumentArray`. These `Executors` will live in different runtime depending on how you want to deploy
```

> your flow. By default, if you are serving your flow locally they will live within processes or threads. Nevertheless, 

Reviewer: Do not mention Thread, the usage is not complete

> because jina is think to be cloud native your flow can easily manage executors that live in containers and that are

Reviewer: Capitalize Jina always

orchestrated by you favorite tools. One of the simplest is `Docker Compose` which is supported out of the box. It is so
easy to deploy your `Flow`'s with `Docker Compose` that we even recommend using it for local development.

Under the hood with one line 
```python
flow.to_docker_compose_yaml('docker-compose.yml')
```

> jina will generate for you a `docker-compose.yml` config files that you can use directly with 

Reviewer: Capitalize Jina

(docker-compose)=
# Jina with Docker Compose

> Jina is a cloud native neural search engine. Therefore, one of the simplest ways of either prototyping or serving in

Reviewer: ```suggestion
Jina is a cloud native neural search framework. Therefore, one of the simplest ways of either prototyping or serving in
```

production is to deploy your `Flow` with `docker-compose`.

Remember that a `Flow` defines complex processing pipelines. A `Flow` is composed of `Executors` which run python code
defined to operate on `DocumentArray`. These `Executors` will live in different runtimes depending on how you want to deploy

> your flow. By default, if you are serving your flow locally they will live within processes. Nevertheless, 

Reviewer: capitalize `Flow`

because Jina is thought to be cloud native your flow can easily manage executors that live in containers and that are
orchestrated by your favorite tools. One of the simplest is `Docker Compose` which is supported out of the box. It is so

> easy to deploy your `Flow`'s with `Docker Compose` that we even recommend using it for local development.

Reviewer: I would not go this far as doing recommendations

f.to_docker_compose_yaml('docker-compose.yml')
```

let's take a look at this config file:
```yaml
version: '3.3'
...
services:

>   encoder-head-0: # # # # # # # # # # # 

Reviewer: Make sure to have the latest version, the name of head has changed

                  #                   #   
  encoder-rep-0:  #   Deployment 1    #
                  #                   #
  encoder-rep-1:  # # # # # # # # # # #

  indexer-head-0: # # # # # # # # # # # 
                  #                   #   
  indexer-rep-0:  #   Deployment 2    #
                  #                   #
  indexer-rep-1:  # # # # # # # # # # #

  gateway: 
    ...
    ports:
    - 8080:8080
```

Here you can see that 7 services will be created. Indeed, first each `Flow` has one entrypoint the `gateway`, then each

> `Deployment` has one `Head` and two instances of the executors (either shard or replicas).

Reviewer: not either or, one for each replica of each shard

Now, you can deploy this Flow to your cluster:

```shell
docker-compose -f docker-compose.yml up
```

### Use your search engine and query your flow

> Now that your flow is up and running in your cloud environment you can query it:

Reviewer: not in your cloud environment, in docker compose, here it is specific

f.to_docker_compose_yaml('docker-compose.yml')
```

let's take a look at this config file:
```yaml
version: '3.3'
...
services:

>   encoder-head:   # # # # # # # # # # # 

Reviewer: No need to explain `Deployment 1` and `Deployment 2` here

    - 8080:8080
```

> Here you can see that 7 services will be created.One for the `gateway` which is the entrypoint of the `Flow`. 

Reviewer: ```suggestion
Here you can see that 7 services will be created. One for the `gateway` which is the entrypoint of the `Flow`. 
```

> Then both of the Executors are wrapped into a `Deployment` .

Reviewer: I think you do not need to talk about `Deployment`, just say that you see X services for each executors, corresponding to a Head, and the replicas ...
